examples/platforms/cc2650/radio.c

```diff
@@ -0,0 +1,1715 @@
+        VerifyOrExit(rfCoreModifySourceMatchEntry(idx, EXT_ADDRESS, true) == CMDSTA_Done, error = kThreadError_Failed);
+    }
+    else
+    {
+        /* we are not running, so we must update the values ourselves */
+        sSrcMatchExtData.srcPendEn[idx / 32] |= (1 << (idx % 32));
+        sSrcMatchExtData.srcMatchEn[idx / 32] |= (1 << (idx % 32));
+    }
+
+exit:
+    return error;
+}
+
+/**
+ * Function documented in platform/radio.h
+ */
+ThreadError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const uint8_t *aExtAddress)
+{
+    ThreadError error = kThreadError_None;
+    (void)aInstance;
+    uint8_t idx;
+    VerifyOrExit((idx = rfCoreFindExtSrcMatchIdx((uint64_t *)aExtAddress)) != CC2650_SRC_MATCH_NONE,
+                 error = kThreadError_NoAddress);
+
+    if (sReceiveCmd.status == ACTIVE || sReceiveCmd.status == IEEE_SUSPENDED)
+    {
+        /* we have a running or backgrounded rx command */
+        VerifyOrExit(rfCoreModifySourceMatchEntry(idx, EXT_ADDRESS, false) == CMDSTA_Done, error = kThreadError_Failed);
+    }
+    else
+    {
+        /* we are not running, so we must update the values ourselves */
+        sSrcMatchExtData.srcPendEn[idx] = 0u;
+        sSrcMatchExtData.srcMatchEn[idx] = 0u;
+        sSrcMatchExtData.srcPendEn[idx / 32] &= ~(1 << (idx % 32));
+        sSrcMatchExtData.srcMatchEn[idx / 32] &= ~(1 << (idx % 32));
+    }
+
+exit:
+    return error;
+}
+
+/**
+ * Function documented in platform/radio.h
+ */
+bool otPlatRadioGetPromiscuous(otInstance *aInstance)
+{
+    (void)aInstance;
+    /* we are promiscuous if we are not filtering */
+    return sReceiveCmd.frameFiltOpt.frameFiltEn == 0;
+}
+
+/**
+ * Function documented in platform/radio.h
+ */
+void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
+{
+    (void)aInstance;
+
+    if (sReceiveCmd.status == ACTIVE || sReceiveCmd.status == IEEE_SUSPENDED)
+    {
+        /* we have a running or backgrounded rx command */
+        /* if we are promiscuous, then frame filtering should be disabled */
+        SuccessOrExit(rfCoreModifyRxFrameFilter(!aEnable) == CMDSTA_Done);
+        /* XXX should we dump any queued messages ? */
+    }
+    else
+    {
+        /* if we are promiscuous, then frame filtering should be disabled */
+        sReceiveCmd.frameFiltOpt.frameFiltEn = aEnable ? 0 : 1;
+    }
+
+exit:
+    return;
+}
+
+/**
+ * Function documented in platform/radio.h
+ */
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    uint8_t *eui64;
+    unsigned int i;
+    bool unknown = true;
+    (void)aInstance;
+
+    /* The IEEE MAC address can be stored two places. We check the Customer
+     * Configuration was not set before defaulting to the Factory
+     * Configuration.
+     */
+    eui64 = (uint8_t *)(CCFG_BASE + CCFG_O_IEEE_MAC_0);
+
+    for (i = 0; i < OT_EXT_ADDRESS_SIZE; i++)
+    {
+        if (eui64[i] != CC2650_UNKNOWN_EUI64)
+        {
+            unknown = false;
```

Comment:
Once you know that this is not an unknown eui, shouldn't you break out of the for loop?